### ISSUE TITLE
LiftCellChat changes expression patterns in shared cell types

### ISSUE THREAD

**Issue by bieniu777** [sqjin/CellChat#247]
Hi, 

I noticed something a bit worrying. 

When I visualise pathway X in my single cellchat object I see it expressed by cell types A and B.  
```
> pathways.show <- c("X") 
> netVisual_heatmap(cellchat_MFTEE_LUSC, signaling = pathways.show, color.heatmap = "Reds")
```

But then, when I run a comparison between three datasets with different cell composition, and I use liftCellChat function on this dataset:
```
group.new = levels(cellchat_MFTEE_LUAD@idents)
cellchat_MFTEE_LUSC <- liftCellChat(cellchat_MFTEE_LUSC, group.new)
```

It shows a diffeent expression of gene X in this dataset. Now it is expressed by cells A and C (and not B anymore), with C expressing this gene being impossible... because these cells do not express this gene at all.

Any ideas on how I could fix that?

BW

P.S. - can it be connected to some of the problems I've been having with merged a cellchat?(https://github.com/sqjin/CellChat/issues/246)

**Comment by sqjin** [comment]
Can you provide a reproducible example as below. Then I will find some time this weekend to check it. 

library(CellChat)
check = readRDS("~/Downloads/sample.RDS")
netVisual_aggregate(check, 
                    targets.use = c("FIB-A","FIB-P"), 
                    signaling = "WNT",
                    layout = "circle")


# Throws error:
# Error in polygon(p[1, ], p[2, ], border = color, lwd = width, lty = lty) : 
#   invalid value specified for graphical parameter "lwd"

**Comment by bieniu777** [comment]
Hi @sqjin 

I don't see this problem when using the data from your paper. I still see it with my data unfortunately. 
But what I noticed is that when creating a heatmap before and after lifting up a cellchat object:

`netVisual_heatmap(cellchat_lusc, signaling = pathways.show, color.heatmap = "Reds")`

the order of my cells changes. In the tutorial data - the new labels are insterted without changing the order of the labels that were initially in the dataset. I however notice that the new labels are added after I lift up my cell chat - but generally all labels - new and old ones - are slightly shuffled. So I am guessing they shouldn't be shuffled and the "wrong" expression that I am seeing is because the labels are placed wrong? 
But I have no idea how to fix that. I have 29 idents in one dataset, 28 in the second and 26 in the third. I lift them both up to the first dataset with 29 idents. I checked them all with 
`levels(cellchat@idents)`
and the names are matching between the datasets - but they do not show up in the same order when running this code. 
e.g. dataset 1:
[1] "Epi9"       "Epi3"         "Epi4"          "Epi1"        "Epi11"        
[6] "Epi7"          "Epi2"         "Epi5"        "Endo1"         "Endo2"     (...)

dataset2:
[1] "Epi3"         "Epi2"         "Epi4"          "Epi9"       "Epi7"         
[6] "Epi1"        "Epi6"        "Epi5"        "Epi11"         "Epi8"      (...)

When running the same on the dataset from the tutorial:
`levels(cellchat.E13@idents)`
[1] "FIB-A"    "FIB-B"    "FIB-P"    "MYL"      "Immune"   "ENDO"     "Muscle"   "MELA"     "Basal-P"  "Basal"    "Spinious"
levels(cellchat.E14@idents)
 [1] "FIB-A"    "FIB-B"    "FIB-P"    "DC"       "Pericyte" "MYL"      "Immune"   "ENDO"     "Muscle"   "MELA"     "Basal-P" 
[12] "Basal"    "Spinious"

They are in the same order.... Just the DC and pericyte are inserted. Exactly as seen on a heatmap.

So I guess the problem must be the order of idents somehow? Is there a way I can re-order them for instance? 

BW
